experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,666 @@
+import Oscar.gens, AbstractAlgebra.direct_sum, Oscar.invariant_ring
+export ReductiveGroup, reductive_group, representation_matrix, group, reynolds_operator, group_ideal, canonical_representation, natural_representation
+export RepresentationReductiveGroup, representation_reductive_group, representation_on_forms, representation_matrix
+export InvariantRing, invariant_ring, fundamental_invariants, null_cone_ideal
+##########################
+#Reductive Groups
+##########################
+mutable struct ReductiveGroup
+    field::Field #characteristic zero. implement check? 
+    group::Tuple{Symbol, Int}
+    group_ideal::MPolyIdeal
+    reynolds_operator::Function
+    canonical_representation::AbstractAlgebra.Generic.MatSpaceElem
+
+    function ReductiveGroup(sym::Symbol, m::Int, fld::Field) #have not decided the representation yet
+        #check char(fld)
+        if sym == :SL 
+            R, _ = polynomial_ring(fld, :z => (1:m,1:m))
+            return ReductiveGroup(sym, m, R)
+        elseif sym == :torus
+            G = new()
+            G.group = (sym, m)
+            characteristic(fld) == 0 || error("Characteristic should be 0 for linearly reductive groups")
+            G.field = fld
+            return G
+        end
+    end
+
+    function ReductiveGroup(sym::Symbol, m::Int, pring::MPolyRing) #the ring input is the group ring
+        #check char(field)
+        G = new()
+        if sym != :SL
+            error("Only implemented for SLm")
+        end
+        fld = base_ring(pring)
+        characteristic(fld) == 0 || error("Characteristic should be 0 for linearly reductive groups")
+        G.field = fld
+        @assert m^2  == ngens(pring)
+        G.group = (sym,m)
+        G.reynolds_operator = reynolds_slm
+        M = matrix(pring, m, m, gens(pring))
+        G.canonical_representation = M
+        G.group_ideal = ideal([det(M) - 1])
+        #base ring of M has to be the same as the representation matrix when that is created later.
+        return G
+    end
+end
+
+function Base.show(io::IO, G::ReductiveGroup)
+    io = AbstractAlgebra.pretty(io)
+    if G.group[1] == :SL
+        print(io, "Reductive group ", G.group[1], G.group[2])
+    else
+        println(io, "Torus of degree ", group(G)[2])
+        print(IOContext(io, :supercompact => true), AbstractAlgebra.Indent(), "over the field ", field(G))
+        print(io, AbstractAlgebra.Dedent())
```

Comment:
```suggestion
        print(IOContext(io, :supercompact => true), Indent(), "over the field ", field(G))
        print(io, Dedent())
```

Reply:
actually, it shouldn't say "over the field" because the field might be printed as "Prime field BLAH" and then you have "over the field Prime field BLAH".

And then there should be a `Lowercase()` inserted, too, to get it to say "prime" not "Prime"
```suggestion
        print(IOContext(io, :supercompact => true), Indent(), "over ", Lowercase(), field(G))
        print(io, Dedent())
```

Reply:
I still see `AbstractAlgebra.Indent` and `AbstractAlgebra.pretty` etc. elsewhere in the file, please make sure to adjust all occurences